src/checker/CidChecker.ts

```diff
@@ -916,6 +923,53 @@ export default class CidChecker {
     return [summary.join('\n'), joinedContent]
   }
 
+  private async providersRetrievability (providerDistributions: ProviderDistributionWithLocation[]): Promise<Array<{ provider_id: string, weightedAvarage: number }>> {
+    try {
+      const to = moment().format('YYYY-MM-DD')
+      const from = moment().subtract(7, 'days').format('YYYY-MM-DD')
+
+      const sparkData = await this.fetchSuccessRate(from, to)
+
+      const matchedProviderDistributions: Array<{
+        miner_id: string
+        success_rate: number
+        total_deal_size: number
+      }> = []
+
+      providerDistributions.forEach(({ provider, total_deal_size }) => {
+        const sparkItem = sparkData.find((x) => x.miner_id === provider)
+        if (sparkItem != null) {
+          matchedProviderDistributions.push({
+            ...sparkItem,
+            total_deal_size: parseInt(total_deal_size)
+          })
+        }
+      })
+
+      const retrievability: Array<{ provider_id: string, weightedAvarage: number }> = matchedProviderDistributions.map(({ miner_id, success_rate, total_deal_size }) => {
+        const weightedSum = success_rate * total_deal_size
+        const weightedAverage = total_deal_size !== 0 ? weightedSum / total_deal_size : 0
+        return {
+          provider_id: miner_id,
+          weightedAvarage: weightedAverage
+        }
+      })
+
+
+      return retrievability
+    } catch (error) {
+      console.error('Failed to fetch data:', error)
+      return []
+    }
+  }
+
+  private async fetchSuccessRate (from: string, to: string): Promise<SparkSuccessRate[]> {
```

Comment:
Bad name - Success rate of what? I suggest:

```suggestion
  private async fetchRetrievalSuccessRate (from: string, to: string): Promise<SparkSuccessRate[]> {
```